src/compat/compat.h

```diff
@@ -711,3 +711,12 @@ initReadOnlyStringInfo(StringInfo str, char *data, int len)
 	str->cursor = 0;
 }
 #endif
+
+/*
+ * PG18 renames ec_derives to ec_derives_list
+ * Add macros so we can use the new naming for older versions.
+ * https://github.com/postgres/postgres/commit/88f55bc9
+ */
+#if PG18_LT
+#define ec_derives_list ec_derives
```

Comment:
this is going to be confusing, let's make it upper case?

Reply:
we have to follow upstream capitalization here otherwise we have to keep the macro forever

Reply:
not really, it can be removed when pg 17 is dropped, at the same time as this macro, but will require code changes

Reply:
the idea is to have our code adopt the pg18 conventions, using different naming from pg18 does not achieve that and i dont think it leads to less confusion rather the opposite